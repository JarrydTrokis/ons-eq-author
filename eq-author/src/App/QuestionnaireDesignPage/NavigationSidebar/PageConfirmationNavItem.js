import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { withRouter } from "react-router-dom";
import CustomPropTypes from "custom-prop-types";
import gql from "graphql-tag";
import { get } from "lodash";

import { buildConfirmationPath } from "utils/UrlUtils";
import NavLink from "./NavLink";
import PlaybackIcon from "./icon-playback.svg?inline";

const StyledPageItem = styled.li`
  padding: 0;
  margin: 0;
  position: relative;
  display: flex;
  align-items: center;
`;

const StyledNavLink = styled(NavLink)`
  padding-left: 1.5em;
`;

export const UnwrappedPageConfirmationNavItem = ({
  questionnaireId,
  page,
  match,
  ...otherProps
}) => {
  const errorCount = get(
    page.confirmation,
    "validationErrorInfo.totalCount",
    0
  );
  return (
    <StyledPageItem data-test="question-confirmation-item" {...otherProps}>
      <StyledNavLink
        to={buildConfirmationPath({
          questionnaireId,
          confirmationId: page.confirmation.id,
          tab: match.params.tab,
        })}
        title={page.confirmation.displayName}
        icon={PlaybackIcon}
        data-test="question-confirmation-link"
        errorCount={errorCount}
      >
        {page.confirmation.displayName}
      </StyledNavLink>
    </StyledPageItem>
  );
};

UnwrappedPageConfirmationNavItem.fragments = {
  PageConfirmationNavItem: gql`
    fragment PageConfirmationNavItem on QuestionPage {
      id
      confirmation {
        id
        displayName
        validationErrorInfo {
          id
          totalCount
        }
      }
    }
  `,
};

UnwrappedPageConfirmationNavItem.propTypes = {
  sectionId: PropTypes.string.isRequired,
  questionnaireId: PropTypes.string.isRequired,
  page: CustomPropTypes.page,
  match: CustomPropTypes.match.isRequired,
};

export default withRouter(UnwrappedPageConfirmationNavItem);
