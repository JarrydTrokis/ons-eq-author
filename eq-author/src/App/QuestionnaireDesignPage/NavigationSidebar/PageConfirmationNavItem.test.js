import React from "react";
import { shallow } from "enzyme";

import { UnwrappedPageConfirmationNavItem as PageConfirmationNavItem } from "./PageConfirmationNavItem";

const createWrapper = props => shallow(<PageConfirmationNavItem {...props} />);

describe("PageConfirmationNavItem", () => {
  let props, wrapper;

  beforeEach(() => {
    props = {
      questionnaireId: "1",
      sectionId: "2",
      page: {
        id: "3",
        confirmation: {
          id: "4",
          displayName: "Confirmation display name",
        },
      },
      match: {
        params: {
          questionnaireId: "1",
          tab: "design",
        },
      },
    };

    wrapper = createWrapper(props);
  });

  it("should render design", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render preview", () => {
    props.match.params.tab = "preview";
    expect(createWrapper({ ...props })).toMatchSnapshot();
  });
});
