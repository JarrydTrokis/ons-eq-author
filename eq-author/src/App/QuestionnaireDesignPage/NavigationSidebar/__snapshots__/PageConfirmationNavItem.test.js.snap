// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageConfirmationNavItem should render design 1`] = `
<PageConfirmationNavItem__StyledPageItem
  data-test="question-confirmation-item"
  sectionId="2"
>
  <PageConfirmationNavItem__StyledNavLink
    data-test="question-confirmation-link"
    errorCount={0}
    icon={[Function]}
    title="Confirmation display name"
    to="/q/1/question-confirmation/4/design"
  >
    Confirmation display name
  </PageConfirmationNavItem__StyledNavLink>
</PageConfirmationNavItem__StyledPageItem>
`;

exports[`PageConfirmationNavItem should render preview 1`] = `
<PageConfirmationNavItem__StyledPageItem
  data-test="question-confirmation-item"
  sectionId="2"
>
  <PageConfirmationNavItem__StyledNavLink
    data-test="question-confirmation-link"
    errorCount={0}
    icon={[Function]}
    title="Confirmation display name"
    to="/q/1/question-confirmation/4/preview"
  >
    Confirmation display name
  </PageConfirmationNavItem__StyledNavLink>
</PageConfirmationNavItem__StyledPageItem>
`;
